server/src/main/java/org/opensearch/index/codec/fuzzy/package-info.java

```diff
@@ -0,0 +1,10 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+/** classes responsible for handling all fuzzy codecs and operations */
+package org.opensearch.index.codec.fuzzy;
```

Comment:
I see that the classes in the `fuzzy` package aren't tied to the `:server` module. May be move them under the `:libs:opensearch-common` module instead?

Reply:
+1, we look into moving to libs

Reply:
1. The placement is done on the basis of the existing structure followed for codecs. The files introduced in the new package are related to new postings format, and the data structure with support for persistence. Both of these rely on Lucene SPIs/APIs which are not available as a dependency in `:libs:opensearch-common`. Are we open to adding the dependency on Lucene jars within the existing lib? Based on the structure, I did not see such files to present in `:libs`.

2. There are few classes like `BigArrays` which these data structures rely on which are part of `:server` module and won't be accessible within the `:libs` Since these abstractions are not accessible, we may require to work/wait for these to be available to be able to move to `:libs` and pick this up in a separate change.